src/xsdata.cpp

```diff
@@ -0,0 +1,700 @@
+    for (int a = 0; a < total.size(); a++) {
+      for (int gin = 0; gin < total[a].size(); gin++) {
+        total[a][gin] += scalar * that->total[a][gin];
+        absorption[a][gin] += scalar * that->absorption[a][gin];
+        if (i == 0) {
+          inverse_velocity[a][gin] = that->inverse_velocity[a][gin];
+        }
+        if (that->prompt_nu_fission.size() > 0) {
+          nu_fission[a][gin] += scalar * that->nu_fission[a][gin];
+          prompt_nu_fission[a][gin] +=
+               scalar * that->prompt_nu_fission[a][gin];
+          kappa_fission[a][gin] += scalar * that->kappa_fission[a][gin];
+          fission[a][gin] += scalar * that->fission[a][gin];
+
+          for (int dg = 0; dg < delayed_nu_fission[a][gin].size(); dg++) {
+            delayed_nu_fission[a][gin][dg] +=
+                 scalar * that->delayed_nu_fission[a][gin][dg];
+          }
+
+          for (int gout = 0; gout < chi_prompt[a][gin].size(); gout++) {
+            chi_prompt[a][gin][gout] +=
+                 scalar * that->chi_prompt[a][gin][gout];
+
+            for (int dg = 0; dg < chi_delayed[a][gin][gout].size(); dg++) {
+              chi_delayed[a][gin][gout][dg] +=
+                   scalar * that->chi_delayed[a][gin][gout][dg];
+            }
+          }
+        }
+      }
+
+      for (int dg = 0; dg < decay_rate[a].size(); dg++) {
+        decay_rate[a][dg] += scalar * that->decay_rate[a][dg];
+      }
+
+      // Normalize chi
+      if (chi_prompt.size() > 0) {
+        for (int gin = 0; gin < chi_prompt[a].size(); gin++) {
+          double norm = std::accumulate(chi_prompt[a][gin].begin(),
+                                        chi_prompt[a][gin].end(), 0.);
+          if (norm > 0.) {
+            for (int gout = 0; gout < chi_prompt[a][gin].size(); gout++) {
+              chi_prompt[a][gin][gout] /= norm;
+            }
+          }
+
+          for (int dg = 0; dg < chi_delayed[a][gin][0].size(); dg++) {
+            norm = 0.;
+            for (int gout = 0; gout < chi_delayed[a][gin].size(); gout++) {
+              norm += chi_delayed[a][gin][gout][dg];
+            }
+            if (norm > 0.) {
+              for (int gout = 0; gout < chi_delayed[a][gin].size(); gout++) {
+                chi_delayed[a][gin][gout][dg] /= norm;
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  // Allow the ScattData object to combine itself
+  for (int a = 0; a < total.size(); a++) {
+    // Build vector of the scattering objects to incorporate
+    std::vector<ScattData*> those_scatts(those_xs.size());
+    for (int i = 0; i < those_xs.size(); i++) {
+      those_scatts[i] = those_xs[i]->scatter[a];
+    }
+
+    // Now combine these guys
+    scatter[a]->combine(those_scatts, scalars);
+  }
+}
+
+//==============================================================================
+
+bool
+XsData::equiv(const XsData& that)
+{
+  bool match = false;
+  if ((absorption.size() == that.absorption.size()) &&
```

Comment:
`return (absorption.size() == that.absorption.size()) && ...`

Reply:
- [x]